hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientMembershipListener.java

```diff
@@ -109,17 +110,15 @@ public void handleMemberListEventV10(Collection<Member> initialMembers) {
     }
 
     @Override
-    public void handleMemberAttributeChangeEventV10(String uuid, String key, int opType, String value) {
-        Collection<Member> members = clusterService.getMemberList();
-        for (Member target : members) {
-            if (target.getUuid().equals(uuid)) {
-                final MemberAttributeOperationType operationType = MemberAttributeOperationType.getValue(opType);
-                ((AbstractMember) target).updateAttribute(operationType, key, value);
-                MemberAttributeEvent memberAttributeEvent =
-                        new MemberAttributeEvent(client.getCluster(), target, operationType, key, value);
-                clusterService.fireMemberAttributeEvent(memberAttributeEvent);
-                break;
-            }
+    public void handleMemberAttributeChangeEventV10(Member member, Collection<Member> members, String key, int opType,
+                                                    String value) {
+        Collection<Member> currentMembersList = clusterService.getMemberList();
+        if (currentMembersList.contains(member)) {
```

Comment:
This causes behavior to change. We need to check only the uuids. Contains use equals which also checks for address equality.

Reply:
Ok, reverted back and opened and opened an issue https://github.com/hazelcast/hazelcast/issues/15271